ledger/src/ui/test/mock/apdu_utils.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 RSK Labs Ltd
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+
+#define CHANNEL_APDU 0
+
+#define IO_APDU_BUFFER_SIZE (5 + 255)
+extern unsigned char G_io_apdu_buffer[IO_APDU_BUFFER_SIZE];
+
+#define ASSERT_APDU(str) \
+    assert(0 == strncmp((const char*)G_io_apdu_buffer, str, strlen(str)))
+#define ASSERT_APDU_AT(i, val) assert(APDU_AT(i) == val)
```

Comment:
These are great. Just a couple of questions:
- Is there really a need for `ASSERT_APDU_AT`? Couldn't we just assert the entirety of the APDU every time?
- Why are we using `#define` as opposed to proper functions? I'm thinking functions tend to be easier to read and modify for e.g. debugging purposes.

Reply:
Good point about `ASSERT_APDU_AT`. My reasoning is that for the purpose of these tests, we don't always need to check the entire APDU buffer, so sometimes it should be useful to check only one specific position. But giving this a second thought, asserting the full buffer every time probably adds completion to the tests. I'll change that.

About using `defines`, I reckon that using functions has several advantages. The only thing that bothers me about using functions is that in the event of an error, the error message will point to the line where `assert()` was actually called, inside the helper function. So any failed assert will point to the same line of `assert_utils.c`. That is less helpful for debugging IMO. When using defines, the error message points to the line where the macro was called, which lets you know exactly which assert failed. What are your thoughts on that?

Reply:
Good point about the lines in the error messages.